cartography/models/aws/ec2/vpc_cidr.py

```diff
@@ -0,0 +1,90 @@
+from dataclasses import dataclass
+
+from cartography.models.core.common import PropertyRef
+from cartography.models.core.nodes import CartographyNodeProperties
+from cartography.models.core.nodes import CartographyNodeSchema
+from cartography.models.core.nodes import ExtraNodeLabels
+from cartography.models.core.relationships import CartographyRelProperties
+from cartography.models.core.relationships import CartographyRelSchema
+from cartography.models.core.relationships import LinkDirection
+from cartography.models.core.relationships import make_target_node_matcher
+from cartography.models.core.relationships import OtherRelationships
+from cartography.models.core.relationships import TargetNodeMatcher
+
+
+@dataclass(frozen=True)
+class AWSIPv4CidrBlockNodeProperties(CartographyNodeProperties):
+    id: PropertyRef = PropertyRef("Id")
+    vpcid: PropertyRef = PropertyRef("VpcId")  # Extra index ?
+    association_id: PropertyRef = PropertyRef("AssociationId")
+    cidr_block: PropertyRef = PropertyRef("CidrBlock")
+    block_state: PropertyRef = PropertyRef("BlockState")
+    block_state_message: PropertyRef = PropertyRef("BlockStateMessage")
+    lastupdated: PropertyRef = PropertyRef("lastupdated", set_in_kwargs=True)
+
+
+@dataclass(frozen=True)
+class AWSIPv4CidrBlockToAWSVpcRelProperties(CartographyRelProperties):
+    lastupdated: PropertyRef = PropertyRef("lastupdated", set_in_kwargs=True)
+
+
+@dataclass(frozen=True)
+class AWSIPv4CidrBlockToAWSVpcRel(CartographyRelSchema):
+    target_node_label: str = "AWSVpc"
+    target_node_matcher: TargetNodeMatcher = make_target_node_matcher(
+        {"id": PropertyRef("VpcId")}
+    )
+    direction: LinkDirection = LinkDirection.INWARD
+    rel_label: str = "BLOCK_ASSOCIATION"
+    properties: AWSIPv4CidrBlockToAWSVpcRelProperties = (
+        AWSIPv4CidrBlockToAWSVpcRelProperties()
+    )
+
+
+@dataclass(frozen=True)
```

Comment:
Could be worth adding a comment saying 

there is no sub resource relationship here because a CIDR block can be associated with more than one account and it doesn't make sense to scope it to one. additionally, scoped_cleanup (https://github.com/cartography-cncf/cartography/blob/072e79419dfbcd2e899a42f3a740816d65659964/cartography/models/core/nodes.py#L108) does not make sense to be overriden here.